kernel/kobject.c

```diff
@@ -9,44 +9,54 @@
 #include "kmalloc.h"
 #include "fs.h"
 #include "device.h"
+#include "library/string.h"
 
 struct kobject *kobject_create_file(struct fs_file *f)
 {
-	struct kobject *k = kmalloc(sizeof(*k));
+	struct kobject *k = kobject_init();
 	k->type = KOBJECT_FILE;
 	k->refcount = 1;
 	k->data.file = f;
 	k->offset = 0;
+	k->intent = 0;
 	return k;
 }
 
 struct kobject *kobject_create_device(struct device *d)
 {
-	struct kobject *k = kmalloc(sizeof(*k));
+	struct kobject *k = kobject_init();
 	k->type = KOBJECT_DEVICE;
-	k->refcount = 1;
 	k->data.device = d;
 	return k;
 }
 
 struct kobject *kobject_create_graphics(struct graphics *g)
 {
-	struct kobject *k = kmalloc(sizeof(*k));
+	struct kobject *k = kobject_init();
 	k->type = KOBJECT_GRAPHICS;
-	k->refcount = 1;
 	k->data.graphics = g;
 	return k;
 }
 
 struct kobject *kobject_create_pipe(struct pipe *p)
 {
-	struct kobject *k = kmalloc(sizeof(*k));
+	struct kobject *k = kobject_init();
 	k->type = KOBJECT_PIPE;
-	k->refcount = 1;
 	k->data.pipe = p;
 	return k;
 }
 
+// Helper function for constant initializations across all types of Kobjects.
+// SHOULD NOT BE CALLED ON ITS OWN, IT DOESN'T GENERATE A VALID KOBJECT.
+struct kobject *kobject_init()
```

Comment:
Make it static, so that it cannot be called outside of the module.

Reply:
My intuition is that it should be private, so it has been removed.`